### ISSUE TITLE
Compose file parsing can't parse Windows paths

### ISSUE THREAD

**Issue by willseward** [docker/docker#31224]
**Description**

The compose file parsing inside Docker can't parse Windows volumes.

Just opening this to track and in case anyone else was experimenting with Swarm and Windows.

Tracking the status of: https://github.com/docker/docker/blob/master/cli/compose/convert/volume.go#L31

**Steps to reproduce the issue:**
1. Create a v3 docker-compose.yml spec with a service.volume spec.
2. Run `docker stack deploy...` 

**Describe the results you received:**
`undefined volume: C`

**Describe the results you expected:**

Volume mounts in a container

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 08:47:51 2017
 OS/Arch:      windows/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 08:47:51 2017
 OS/Arch:      windows/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 2
Server Version: 1.13.1
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
Swarm: active
 NodeID: ov0y3pqc1gjx15upv9mjru6bv
 Is Manager: true
 ClusterID: vkznxg9rw3dmput8jl9k291fz
 Managers: 1
 Nodes: 2
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 172.16.58.181
 Manager Addresses:
  172.16.58.181:2377
Default Isolation: hyperv
Kernel Version: 10.0 15031 (15031.0.amd64fre.rs2_release.170204-1546)
Operating System: Windows 10 Enterprise Insider Preview
OSType: windows
Architecture: x86_64
CPUs: 1
Total Memory: 3 GiB
Name: DESKTOP-UJ1M2UA
ID: 5JUS:Y5TR:DAEX:KQ7U:SC3Z:3IBV:3HBH:XNN5:K7SP:JDL6:J2OB:6FQR
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 131
 System Time: 2017-02-21T08:17:39.3966974-08:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by friism** [comment]
cc @aluzzardi @vsaraswat @mgoelzer

**Comment by aluzzardi** [comment]
/cc @dnephin 

Volumes names cannot be <3 characters and Windows volumes cannot be >1 characters so this should be pretty easy to catch

**Comment by dnephin** [comment]
I believe this is fixed by #30597, which could use some attention, it's been open for a month or so

**Comment by git-jiby-me** [comment]
@dnephin @thaJeztah is it possible, that we can extract just the windows path parsing logic into a different smaller simple PR, so that it can be merged faster.

Currently the parsing issue completely blocks people from using volumes in windows nodes in a swarm, which IMHO is a big blocker bug.

**Comment by dnephin** [comment]
#30597  should be in the next release, so there isn't really any faster option

**Comment by git-jiby-me** [comment]
@dnephin ok, good to know, if this is slated for next release, then thats good enough for me.

**Comment by artisticcheese** [comment]
Is this fixed in current release? Getting the same error.

**Comment by chairbender** [comment]
I'm getting this issue on the latest release.
`version: '3'
services:
  webservice:
    image: tomcat:8.5
    ports:
      - "8080:8080"
    volumes:
      - ./my-webapp.war:/usr/local/tomcat/webapps/my-webapp.war`

My docker version:
`Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Fri Mar 24 00:00:50 2017
 OS/Arch:      linux/amd64
 Experimental: true
`

**Comment by thaJeztah** [comment]
17.03.1 is the latest release in the _stable_ channel (quarterly releases), but this issue is fixed in Docker 17.04; https://github.com/moby/moby/pull/30597.

Current "edge" release is 17.05, but the next stable/quarterly release is 17.06 and will be released soon.

**Comment by artisticcheese** [comment]
When "soon" is coming?
